deploy.tf

```diff
@@ -381,10 +374,154 @@ EOF
 
 ###############################################################################
 #
-# Make config file and export variables for kubectl
+# etcd host firewall
+#
+###############################################################################
+
+resource "digitalocean_firewall" "k8s_etcd" {
+  name = "${var.prefix}k8s-etcd-fw"
+
+  droplet_ids = ["${digitalocean_droplet.k8s_etcd.id}"]
+
+  inbound_rule = [
+    {
+      protocol           = "tcp"
+      port_range         = "22"
+      source_addresses   = ["${digitalocean_droplet.k8s_master.ipv4_address_private}"]
+    },
+    {
+      protocol           = "tcp"
+      port_range         = "2379-2380"
+      source_addresses   = ["${digitalocean_droplet.k8s_master.ipv4_address_private}", "${digitalocean_droplet.k8s_worker.*.ipv4_address_private}"]
+    },
+  ]
+
+  outbound_rule = [
+    {
+      protocol                = "tcp"
+      port_range              = "all"
+      destination_addresses   = ["0.0.0.0/0", "::/0"]
+    },
+    {
+      protocol                = "udp"
+      port_range              = "all"
+      destination_addresses   = ["0.0.0.0/0", "::/0"]
+    },
+  ]
+
+}
+
+###############################################################################
+#
+# worker host firewall
+#
+###############################################################################
+
+resource "digitalocean_firewall" "k8s_worker" {
+  name = "${var.prefix}k8s-worker-fw"
+
+  droplet_ids = ["${digitalocean_droplet.k8s_worker.*.id}"]
+
+  inbound_rule = [
+    {
+      protocol           = "tcp"
+      port_range         = "all"
+      source_addresses   = ["${digitalocean_droplet.k8s_master.ipv4_address_private}", "${digitalocean_droplet.k8s_worker.*.ipv4_address_private}"]
+    },
+    {
+      protocol           = "tcp"
+      port_range         = "10250"
+      source_addresses   = ["${digitalocean_droplet.k8s_master.ipv4_address_private}", "${digitalocean_droplet.k8s_worker.*.ipv4_address_private}"]
+    },
+    {
+      protocol           = "tcp"
+      port_range         = "10255"
```

Comment:
I found a sticky spot here where port 10255 should be open not from master but from which ever node heapster is running on. However we don't really know which node if any heapster will be running on. Since running heapster on master would require a special annotation per #43 to run it on master *maybe* we can assume that it would mostly likely appear on a worker and open 10255 from all worker nodes. Or maybe it would be a better idea just to leave it closed. Any thoughts?

Reply:
I would say leaving it open to master only is reasonable

Reply:
Sorry for the delay. I pushed some comments for the firewall rules in 30024bbc.